### ISSUE TITLE
Document process of setting up the project for development

### ISSUE THREAD

**Issue by sonukushwaha403** [JOSM/pt_assistant#27]
updating the Readme file with Supplemental information for JOSM .

**Comment by sonukushwaha403** [comment]
i have created a pr it shows that I need reviewer to be reviewed  [here](https://github.com/JOSM/pt_assistant/pull/29)

**Comment by PolyglotOpenstreetmap** [comment]
Hi Sonu,

Florian will probably review it, but it might take a few days.

Jo

**Comment by floscher** [comment]
@sonukushwaha403 You don't necessarily need to have JOSM installed for development. We use Gradle as build tool and it will provide you with the correct JOSM version.

With the Gradle task
```bash
./gradlew runJosm
```
you can start JOSM with the current state of the plugin already loaded. Some other useful tasks are `assemble` to compile, `test` for unit tests, `dist` for creating the *.jar distribution, …

If you need help setting up the IDE or what is where in the code, feel free to ask here. Maybe we can write down the steps for setting things up for the future, as something like https://github.com/JOSM/Mapillary/blob/master/INSTALL.md (that one is already a bit outdated by the way).

**Comment by floscher** [comment]
@sonukushwaha403 were you able to set up the project for development?
If not, do you need assistance? If yes, would you like to write down the steps of the process?

**Comment by sonukushwaha403** [comment]
Yes, I need assistance to set up, please guide me..

**Comment by floscher** [comment]
What IDE are you using? In IntelliJ idea it should work to just open the project directory.
Are the Gradle tasks working that I wrote above? There shouldn't really be setup otherwise, except having a JDK >=8 installed.
Maybe try `./gradlew runJosm` from the command line (or run the `runJosm` task from the IDE) which should get the dependencies, build the plugin and runs a JOSM instance with the built plugin loaded. Does that work?

@PolyglotOpenstreetmap did you have to take other steps when you set up the project recently?

**Comment by PolyglotOpenstreetmap** [comment]
Hi Sonu and Florian,

I did it twice recently (I somehow mangled up the first time by switching
between branches).

All I had to do was get the https link on github for the repository (
https://github.com/JOSM/pt_assistant.git) and tell IntelliJ to download it
using

File/New Project from version control...

Then I went to VCS/Git/Branches... but that's optional, I think.

To run it, I open Gradle on the right side, Tasks/josm/runJosm. After the
first time, I can simply use the green play button in the top bar, or the
green bug button (after setting a breakpoint where I want it to stop, to
figure out what bugs I introduced this time... :-) ).

Jo
